checks/raw/sbom.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raw
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/ossf/scorecard/v5/checker"
+	"github.com/ossf/scorecard/v5/clients"
+	"github.com/ossf/scorecard/v5/finding"
+)
+
+var (
+	reRootFile = regexp.MustCompile(`^[^.]([^//]*)$`)
+	reSBOMFile = regexp.MustCompile(
+		`(?i).+\.(cdx.json|cdx.xml|spdx|spdx.json|spdx.xml|spdx.y[a?]ml|spdx.rdf|spdx.rdf.xm)`,
+	)
+)
+
+// SBOM retrieves the raw data for the SBOM check.
+func SBOM(c *checker.CheckRequest) (checker.SBOMData, error) {
+	var results checker.SBOMData
+
+	releases, lerr := c.RepoClient.ListReleases()
+	if lerr != nil {
+		return results, fmt.Errorf("RepoClient.ListReleases: %w", lerr)
+	}
+
+	results.SBOMFiles = append(results.SBOMFiles, checkSBOMReleases(releases)...)
+
+	// Look for SBOMs in source
+	repoFiles, err := c.RepoClient.ListFiles(func(string) (bool, error) { return true, nil })
```

Comment:
this will match every file in the repo and put it in a slice. I see you filter it with `checkSBOMSource` down lower, but it would be simpler to change your `predicate` func to do the filtering here instead of later: 

```go
// Look for SBOMs in source
repoFiles, err := c.RepoClient.ListFiles(func(string) (bool, error) { 
    return reSBOMFile.MatchString(file) && reRootFile.MatchString(file), nil 
})
```

You'll still need to iterate to create the `checker.SBOM`, but we reduce the allocations needed enumerating files